src/rgw/CMakeLists.txt

```diff
@@ -178,10 +178,6 @@ if (WITH_RADOSGW_BEAST_FRONTEND)
   target_link_libraries(rgw_a Boost::coroutine Boost::context)
 endif()
 
-if (WITH_CURL_OPENSSL OR (WITH_RADOSGW_BEAST_FRONTEND AND WITH_RADOSGW_BEAST_OPENSSL))
-  target_link_libraries(rgw_a ${OPENSSL_LIBRARIES})
```

Comment:
this change seems to drop the `WITH_CURL_OPENSSL` condition

@theanalyst @mattbenjamin  does anyone remember the specifics here? ie which platforms use libcurl+openssl, and whether we need to link openssl explicitly? the [shaman builds](https://shaman.ceph.com/builds/ceph/wip-pr-24368/3674c1cc64a6e1aa34291d200f1c904852a3ed3a/) all succeeded, so i'm tempted to ack this

Reply:
@cbodley the reason i made this change is based on the inspection on `rgw_asio_frontend.cc`, it pulls in `boost/asio/ssl.hpp` if `WITH_RADOSGW_BEAST_OPENSSL` is defined. so i think we should link against `OpenSSL::SSL` as long as `WITH_RADOSGW_BEAST_FRONTEND` and `WITH_RADOSGW_BEAST_OPENSSL`.

while `rgw_http_client_curl.cc` pulls in `<openssl/crypto.h>` if `WITH_CURL_OPENSSL` is defined, so in that case, we need to link against `OpenSSL::Crypto` i think.

probably, i need to update this part like
```cmake
if(WITH_CURL_OPENSSL)
  # used by rgw_http_client_curl.cc
  target_link_libraries(radosgw_a PRIVATE OpenSSL:: Crypto)
endif()
if(WITH_RADOSGW_BEAST_FRONTEND AND WITH_RADOSGW_BEAST_OPENSSL)
  # used by rgw_asio_frontend.cc
  target_link_libraries(radosgw_a PRIVATE OpenSSL::SSL)
endif()
```

instead of linking OpenSSL libraries as a whole. does this make more sense?

Reply:
i like that @tchaikov, thanks - i think the WITH_CURL_OPENSSL case should apply to rgw_a instead though, so it covers the other targets - then you can remove ${SSL_LIBRARIES} from radosgw-admin and radosgw-es

Reply:
yeah agree with @cbodley here, the changeset proposed looks good, but we'll need rgw_a as librgw, rgw_admin also call into curl